api/controllers/v0.2/createPatron.js

```diff
@@ -55,74 +55,155 @@ function renderResponse(req, res, status, message) {
-        ) // eslint-disable-line comma-dangle
-      ));
+function streamPatron(req, res, streamPatronData, modeledResponse) {
+  streamPublish.streamPublish(
+    process.env.PATRON_SCHEMA_NAME_V02,
+    process.env.PATRON_STREAM_NAME_V02,
+    streamPatronData // eslint-disable-line comma-dangle
+  ).then((streamResponse) => { // eslint-disable-line no-unused-vars
+    renderResponse(req, res, 201, modeledResponse);
+    console.log('Published to stream successfully!'); // eslint-disable-line no-console
+  }).catch((streamError) => {
+    renderResponse(req, res, 201, streamError);
+    console.error(`Error publishing to stream: ${streamError}`); // eslint-disable-line no-console
+  });
+}
 
-    return;
-  }
+/**
+ * callAxiosToCreatePatron(req, res, tokenResponse)
+ * Stream the new patron data to Kinesis
+ *
+ * @param {HTTP request} req
+ * @param {HTTP response} res
+ * @param {object} tokenResponse
+ */
+function callAxiosToCreatePatron(req, res, tokenResponse) {
+  axios.post(process.env.ILS_CREATE_PATRON_URL, req.body, {
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${tokenResponse.data.access_token}`,
+    },
+  })
+    .then((response) => {
+      const modeledResponse =
+        modelResponse.patronCreator(response.data, response.status, req.body);
+      modelStreamPatron.transformPatronRequest(
+        req.body, modeledResponse // eslint-disable-line comma-dangle
+      )
+        .then((streamPatronData) => {
+          streamPatron(req, res, streamPatronData, modeledResponse);
+        })
+        .catch((error) => {
+          renderResponse(req, res, 201, modeledResponse);
+          console.error(`Error creating patron: ${error}`); // eslint-disable-line no-console
+        });
+    })
+    .catch((response) => {
+      // eslint-disable-next-line no-console
+      console.error(
+        `status_code: ${response.response.status}, ` +
+        'type: "invalid-request", ' +
+        `message: "${response.message} from ILS.", ` +
+        `response: ${JSON.stringify(response.response.data)}` // eslint-disable-line comma-dangle
+      );
+
+      if (response.response && response.response.data) {
+        const responseObject = collectErrorResponseData(
+          response.response.status,
+          response.response.data.type,
+          response.response.data.detail,
+          response.response.data.title,
+          response.response.data.debug_message // eslint-disable-line comma-dangle
+        );
+
+        const statusCode = (responseObject.status) ? responseObject.status : 500;
+
+        renderResponse(
+          req,
+          res,
+          statusCode,
+          modelResponse.errorResponseData(responseObject) // eslint-disable-line comma-dangle
+        );
+      } else {
+        renderResponse(req, res, 500, modelResponse.errorResponseData(
+          collectErrorResponseData(null, '', '', '', '') // eslint-disable-line comma-dangle
```

Comment:
Is it reasonable to send an error response with no data inside?  (I didn't trace all the code logic, so am not sure if this is your absolute last resort.)

Reply:
@daryanypl I'm picking up the established pattern from v0.1.  If it should change, let's make this a tech debt card on Jira.

Reply:
ok with me